addons/mail/static/src/webclient/web/webclient.js

```diff
@@ -62,10 +64,25 @@ patch(WebClient.prototype, {
         // This may occur if the subscription was refreshed by the browser,
         // but it may also happen if the subscription has been revoked or lost.
         if (!subscription) {
-            subscription = await pushManager.subscribe({
-                userVisibleOnly: true,
-                applicationServerKey: await this._getApplicationServerKey(),
-            });
+            try {
+                subscription = await pushManager.subscribe({
+                    userVisibleOnly: true,
+                    applicationServerKey: await this._getApplicationServerKey(),
+                });
+            } catch (error) {
+                console.warn(error);
+                if ((navigator.brave && (await navigator.brave.isBrave())) || false) {
+                    return this.notification.add(_t("Enable Google Services for Push Messaging"), {
+                        title: _t("Failed to activate desktop notifications"),
+                        type: "danger",
+                        sticky: true,
+                    });
+                }
+                return this.notification.add(_t("Failed to activate desktop notifications"), {
+                    type: "danger",
+                    sticky: true,
+                });
```

Comment:
- Would clarify instruction as follow:
_"Enable 'Google Services for Push Messaging' to enable push notifications"_
- Usually we say "enable push notifications" rather than "activate". It works but it sounds less common / natural
- Couldn't we provide `error.reason` or equivalent? Ok to say there was a failure but it could be frustrating to not know why. Even if this is technical (wording and/or with a code error), but at the very least there's something that can help user to investigate and solve the issue. This isn't the core fix of PR so I wouldn't mind to not improve this aspect further in this PR.